src/main/scala/nl/knaw/dans/easy/stage/EasyStageDataset.scala

```diff
@@ -121,9 +144,8 @@ object EasyStageDataset {
         fileAccessRights <- getFileAccessRights(fileMetadata)
         fis = FileItemSettings(
           sdoSetDir = s.sdoSetDir,
-          file = if (s.stageFileDataAsRedirectDatastreams) None else Some(file),
-          datastreamLocation = if (s.stageFileDataAsRedirectDatastreams) s.fileDataRedirectBaseUrl.map(baseUrl => new URL(baseUrl, urlEncodedDatasetRelativePath.toString))
-                               else None,
+          file = if (s.fileUris.get(getBagRelativePath(file.toPath)).isDefined) None else Some(file),
```

Comment:
@rvanheest , @jo-pol Reversed this one. Note that the original lines return `None` *if the entry is defined*.